ConfigModule for _{dynamo,inductor}.config

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #93878
* #93871
* #93870
* #93863
* #93822
* #93821
* #93818
* #93409
* #93408
* #93389
* #93382
* #93381
* #93279
* #93278
* #93254
* #93375
* #93253
* __->__ #93252

This refactors the way dynamo/inductor configs are handled to check for invalid configs and add options like patching and serialization.


cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

jansel committed:
ConfigModule for _{dynamo,inductor}.config

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"




cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

voznesenskym commented on torch/_dynamo/logging.py:
```diff
@@ -25,6 +25,10 @@ def set_loggers_level(level):
         logger.setLevel(level)
 
 
+def get_loggers_level():
+    return get_loggers()[0].level
```
quick comment on what this is?

voznesenskym commented on torch/__init__.py:
```diff
@@ -1308,20 +1308,60 @@ def compiled_with_cxx11_abi():
+        self.config = dict()
+        self.apply_mode(mode)
+        self.apply_passes(passes)
+        if dynamic:
+            # cudagraphs conflicts with dynamic shapes
+            self.config["triton.cudagraphs"] = False
```
Instead of silently contradicting the config, it feels like we should assert and force users to set it False

voznesenskym commented on torch/__init__.py:
```diff
@@ -1308,20 +1308,60 @@ def compiled_with_cxx11_abi():
+            # cudagraphs conflicts with dynamic shapes
+            self.config["triton.cudagraphs"] = False
+
+    def apply_mode(self, mode):
```
what is None? why not make default default and throw out the optionality?

jansel replied:
This is preserving the old behavior (either passes or mode can be set but not both).

voznesenskym commented on torch/__init__.py:
```diff
@@ -1308,20 +1308,60 @@ def compiled_with_cxx11_abi():
+                f"Unrecognized mode={mode}, should be one of: default, reduce-overhead, max-autotune"
+            )
+
+    def apply_passes(self, passes):
```
nit: type annotations on passes would be of use

voznesenskym reviewed (approved):
neat

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and add options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and add options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and add options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel commented:
@pytorchbot merge

jansel commented:
@pytorchbot merge

jansel committed:
Update on "ConfigModule for _{dynamo,inductor}.config"


This refactors the way dynamo/inductor configs are handled to check for invalid configs and add options like patching and serialization.


cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel commented:
@pytorchbot merge
